client.go

```diff
@@ -50,43 +46,61 @@ func (c *Client) Get(ctx context.Context, req *Request) (*GetResult, error) {
 		req.Mode = ModeAny
 	}
 
-	var err error
-	req.Src, err = Detect(req.Src, req.Pwd, c.Detectors)
-	if err != nil {
-		return nil, err
-	}
-
-	var force string
-	// Determine if we have a forced protocol, i.e. "git::http://..."
-	force, req.Src = getForcedGetter(req.Src)
-
 	// If there is a subdir component, then we download the root separately
 	// and then copy over the proper subdir.
-	var realDst, subDir string
-	req.Src, subDir = SourceDirSubdir(req.Src)
-	if subDir != "" {
+	req.Src, req.subDir = SourceDirSubdir(req.Src)
+	if req.subDir != "" {
 		td, tdcloser, err := safetemp.Dir("", "getter")
 		if err != nil {
 			return nil, err
 		}
 		defer tdcloser.Close()
 
-		realDst = req.Dst
+		req.realDst = req.Dst
 		req.Dst = td
 	}
 
-	req.u, err = urlhelper.Parse(req.Src)
-	if err != nil {
-		return nil, err
+	var multierr []error
+	for _, g := range c.Getters {
+		ok, err := Detect(req, g)
+		if err != nil {
+			return nil, err
+		}
+		if !ok {
+			continue
+		}
+
+		result, fatal, err := c.get(ctx, req, g)
+		if err != nil {
+			if fatal {
+				return nil, err
+			}
+			multierr = append(multierr, err)
+			continue
+		}
+
+		return result, nil
 	}
-	if force == "" {
-		force = req.u.Scheme
+
+	if len(multierr) == 1 {
+		// This is for keeping the error original format
+		return nil, multierr[0]
 	}
 
-	g, ok := c.Getters[force]
-	if !ok {
-		return nil, fmt.Errorf(
-			"download not supported for scheme '%s'", force)
+	if multierr != nil {
+		var result *multierror.Error
+		result = multierror.Append(result, multierr...)
+		return nil, fmt.Errorf("error downloading '%s': %s", req.Src, result.Error())
+	}
+
+	return nil, fmt.Errorf("error downloading '%s'", req.Src)
+}
+
+func (c *Client) get(ctx context.Context, req *Request, g Getter) (*GetResult, bool, error) {
```

Comment:
This returns a bool to tell if the error is fatal or not. 
This is because some errors are fatal, I mean when something is wrong with the source string, for these we return `nil, true, err`. Besides these errors, is also possible that the actual download will fail inside the getter and when this happens we want to try any other possible getter in the getters list, so we return `nil, false, err` and save the error to return later if the download is not successful at all.

Reply:
It feels like this bool is here to tell when the calling loop needs to stop.

May `get` could return a specific error:

```go
type getError struct {
	// when fatal is true it means that something when wrong with
	// the get and that the getter knows nothing more should be tried
	Fatal	bool
	Error	error
}

func (ge *GetError) Error() string {
	// ...
}
```

```suggestion
func (c *Client) get(ctx context.Context, req *Request, g Getter) (*GetResult, *getError) {
```